Source/DafnyCore/CompileNestedMatch/MatchFlattener.cs

```diff
@@ -49,23 +49,14 @@ public MatchFlattener(ErrorReporter reporter, FreshIdGenerator idGenerator)
     this.idGenerator = idGenerator;
   }
 
-  internal override void PostResolve(Program program) {
-    foreach (var compileModule in program.RawModules()) {
-      FlattenNode(compileModule);
-    }
-    foreach (var compileModule in program.CompileModules) {
```

Comment:
Since the PR that removed compile modules, the second foreach has become obsolete, setting the `Flattened` property of nested matches when it's already set.

Reply:
Because I'm using `PostResolve(ModuleDefinition` instead of `PostResolve(Program`, this transformation runs slightly earlier now and some integration tests had to be updated because diagnostics from this and another transformation have changed order.